docs/sphinx/source/whatsnew/v0.10.4.rst

```diff
@@ -11,7 +11,8 @@ Enhancements
 
 Bug fixes
 ~~~~~~~~~
-
+* Removed `or` clause from `if` statement in `infer_temperature_model` in
+`ModelChain` (:issue:`1759`).
```

Comment:
```suggestion
* :py:class:`~pvlib.modelchain.ModelChain` now raises a more useful error when
  ``temperature_model_parameters`` are specified on the passed ``system`` instead of on its ``arrays``. (:issue:`1759`).
```